File: TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/format/inread/InReadScrollViewFragment.java
package tv.teads.teadssdkdemo.format.inread;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Toast;

import org.greenrobot.eventbus.Subscribe;

import tv.teads.sdk.android.AdFailedReason;
import tv.teads.sdk.android.InReadAdView;
import tv.teads.sdk.android.TeadsListener;
import tv.teads.teadssdkdemo.R;
import tv.teads.teadssdkdemo.utils.BaseFragment;
import tv.teads.teadssdkdemo.utils.ReloadEvent;

/**
 * InRead format within a ScrollView
 * <p/>
 * Created by Hugo Gresse on 30/03/15.
 */
public class InReadScrollViewFragment extends BaseFragment {

    /**
     * Teads Ad view
     */
    private InReadAdView mAdView;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        View rootView = inflater.inflate(R.layout.fragment_inread_scrollview, container, false);
        mAdView = rootView.findViewById(R.id.teads_ad_view);
        return rootView;
    }

    @Override
    public void onViewCreated(@NonNull View view, Bundle savedInstanceState) {

        // Instanciate Teads Ad in inReadTop format
        mAdView.setPid(getPid());
        mAdView.setListener(mTeadsListener);
        mAdView.load();
    }

    @Override
    public void onDestroy() {
        super.onDestroy();

        if (mAdView != null) {
            mAdView.clean();
        }
    }

    @Subscribe
    @SuppressWarnings("unused")
    public void onReloadEvent(ReloadEvent event) {
        if (mAdView != null) {
            mAdView.load();
        }
    }

    private TeadsListener mTeadsListener = new TeadsListener(){

        @Override
        public void onAdFailedToLoad(AdFailedReason adFailedReason) {
            Toast.makeText(InReadScrollViewFragment.this.getActivity(), getString(R.string.didfail), Toast.LENGTH_SHORT).show();
        }

        @Override
        public void onError(String s) {
            Toast.makeText(InReadScrollViewFragment.this.getActivity(), getString(R.string.didfail_playback), Toast.LENGTH_SHORT).show();
        }
    };

}

File: TeadsSDKDemo/app/src/main/java/tv/teads/teadssdkdemo/format/inread/InReadScrollViewFragment.kt
package tv.teads.teadssdkdemo.format.inread

import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import tv.teads.sdk.AdPlacementSettings
import tv.teads.sdk.AdRequestSettings
import tv.teads.sdk.InReadAdPlacement
import tv.teads.sdk.TeadsSDK
import tv.teads.sdk.combinedsdk.TeadsAdPlacementEventName
import tv.teads.sdk.combinedsdk.adplacement.TeadsAdPlacementMedia
import tv.teads.sdk.combinedsdk.adplacement.config.TeadsAdPlacementMediaConfig
import tv.teads.sdk.combinedsdk.adplacement.interfaces.TeadsAdPlacementEventsDelegate
import tv.teads.sdk.combinedsdk.adplacement.interfaces.core.TeadsAdPlacement
import tv.teads.sdk.renderer.InReadAdView
import tv.teads.teadssdkdemo.R
import tv.teads.teadssdkdemo.databinding.FragmentInreadScrollviewBinding
import tv.teads.teadssdkdemo.utils.BaseFragment

/**
 * InRead format within a ScrollView
 */
class InReadScrollViewFragment : BaseFragment() {
    private lateinit var binding: FragmentInreadScrollviewBinding
    private lateinit var adPlacement: InReadAdPlacement
    private var inReadAdView: InReadAdView? = null

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View {
        binding = FragmentInreadScrollviewBinding.inflate(layoutInflater)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        view.findViewById<TextView>(R.id.integration_header).text = getTitle()

        // 1. Setup the settings
        val placementSettings = AdPlacementSettings.Builder()
            .enableDebug()
            .build()

        val config = TeadsAdPlacementMediaConfig(pid = pid)

        // 2. Create the InReadAdPlacement
        adPlacement = TeadsSDK.createInReadPlacement(requireActivity(), pid, placementSettings)

        val placement = TeadsAdPlacementMedia(
            requireContext(), config, object : TeadsAdPlacementEventsDelegate {
                override fun onPlacementEvent(
                    placement: TeadsAdPlacement<*, *>,
                    event: TeadsAdPlacementEventName,
                    data: Map<String, Any>?
                ) {
                    Log.d("onPlacementEvent", "$event $data")
                }

            }
        )



        // 3. Request the ad and listen its events
        val requestSettings = AdRequestSettings.Builder()
            .pageSlotUrl("http://teads.com")
            .build()

//        adPlacement.requestAd(requestSettings,
//            object : InReadAdViewListener {
//                override fun onAdReceived(ad: InReadAdView, adRatio: AdRatio) {
//                     Clean and init inReadAdView
//                    inReadAdView?.clean()
//                    inReadAdView = ad
//                     Add ad to the container and resize
//                    binding.adSlotContainer.addView(ad)
//                    binding.adSlotContainer.resizeAdContainer(adRatio)
//                }
//
//                override fun adOpportunityTrackerView(trackerView: AdOpportunityTrackerView) {
//                    binding.adSlotContainer.addView(trackerView)
//                }
//
//                override fun onAdRatioUpdate(adRatio: AdRatio) {
//                     Resize the ad container
//                    binding.adSlotContainer.resizeAdContainer(adRatio)
//                }
//
//                override fun onAdClicked() {}
//                override fun onAdClosed() {}
//                override fun onAdError(code: Int, description: String) {}
//                override fun onAdImpression() {}
//                override fun onAdExpandedToFullscreen() {}
//                override fun onAdCollapsedFromFullscreen() {}
//                override fun onFailToReceiveAd(failReason: String) {}
//            },
//            object : VideoPlaybackListener {
//                override fun onVideoComplete() {
//                    Log.d("PlaybackEvent", "complete")
//                }
//
//                override fun onVideoPause() {
//                    Log.d("PlaybackEvent", "pause")
//                }
//
//                override fun onVideoPlay() {
//                    Log.d("PlaybackEvent", "play")
//                }
//
//            }
//        )

        val adView = placement.loadAd()
        binding.adSlotContainer.addView(adView)
    }

    override fun onDestroy() {
        super.onDestroy()
        inReadAdView?.clean()
    }

    override fun getTitle(): String = "InRead Direct ScrollView"
}
